### ISSUE TITLE
UnboundLocalError: local variable 'from_address' referenced before assignment

### ISSUE THREAD

**Issue by atkawa7** [eventlet/eventlet#433]
I am running celery and I am having problems with  the  above error at this line
https://github.com/eventlet/eventlet/blob/ce72c4378b176cdb0d8b6039023e3ad0026becde/eventlet/support/greendns.py#L714

**Comment by temoto** [comment]
Thank you, it's clear.

**Comment by temoto** [comment]
Please, try this patch.

`pip install https://github.com/eventlet/eventlet/archive/8b88cbebe5cc0463b7531b59dc255e36b9e498df.zip`

**Comment by atkawa7** [comment]
@temoto works like a charm. Thanks

![image](https://user-images.githubusercontent.com/5673828/29692119-4d7c4ea6-88e3-11e7-88fa-b61d957198df.png)

You can go ahead and close the issue at your convenience

**Comment by temoto** [comment]
Fix merged in master 8b88cbebe5cc0463b7531b59dc255e36b9e498df

**Comment by MPagel** [comment]
I assume this (or a previous fix) fixes a similar error

> Unexpected error in process <type 'exceptions.AttributeError'>
> Traceback (most recent call last):
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\hubs\hub.py", line 458, in fire_timers
>     timer()
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\hubs\timer.py", line 58, in __call__
>     cb(*args, **kw)
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\hubs\__init__.py", line 155, in _timeout
>     current.throw(exc)
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\greenthread.py", line 218, in main
>     result = function(*args, **kwargs)
>   File "ReadQueues.py", line 821, in doQueueDownload
>     q_obj = returnOneQueueObj(sect)
>   File "ReadQueues.py", line 470, in returnOneQueueObj
>     q = cx.get_queue(qn)
>   File "C:\Program Files\Python\27\lib\site-packages\boto\sqs\connection.py", line 521, in get_queue
>     return self.get_object('GetQueueUrl', params, Queue)
>   File "C:\Program Files\Python\27\lib\site-packages\boto\connection.py", line 1192, in get_object
>     response = self.make_request(action, params, path, verb)
>   File "C:\Program Files\Python\27\lib\site-packages\boto\connection.py", line 1116, in make_request
>     return self._mexe(http_request)
>   File "C:\Program Files\Python\27\lib\site-packages\boto\connection.py", line 943, in _mexe
>     request.body, request.headers)
>   File "C:\Program Files\Python\27\lib\httplib.py", line 1042, in request
>     self._send_request(method, url, body, headers)
>   File "C:\Program Files\Python\27\lib\httplib.py", line 1082, in _send_request
>     self.endheaders(body)
>   File "C:\Program Files\Python\27\lib\httplib.py", line 1038, in endheaders
>     self._send_output(message_body)
>   File "C:\Program Files\Python\27\lib\httplib.py", line 882, in _send_output
>     self.send(msg)
>   File "C:\Program Files\Python\27\lib\httplib.py", line 844, in send
>     self.connect()
>   File "C:\Program Files\Python\27\lib\site-packages\boto\https_connection.py", line 119, in connect
>     sock = socket.create_connection((self.host, self.port), self.timeout)
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\green\socket.py", line 44, in Create_connection
>     for res in getaddrinfo(host, port, 0, SOCK_STREAM):
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\support\greendns.py", line 502, in getaddrinfo
>     qname, addrs = _getaddrinfo_lookup(host, family, flags)
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\support\greendns.py", line 466, in _getaddrinfo_lookup
>     answer = resolve(host, qfamily, False)
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\support\greendns.py", line 413, in resolve
>     return _proxy.query(name, rdtype, raise_on_no_answer=raises)
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\support\greendns.py", line 371, in query
>     return end()
>   File "C:\Program Files\Python\27\lib\site-packages\eventlet\support\greendns.py", line 350, in end
>     raise result[1]
> UnboundLocalError: local variable 'from_address' referenced before assignment

> STACKTRACE:
> Locals by frame, innermost last
> Frame run in C:\Program Files\Python\27\lib\site-packages\eventlet\hubs\hub.py at line 356
>         a: List (size 0): ()
>         wakeup_when: 352173.859
>         self: <eventlet.hubs.selects.Hub object at 0x0000000003C8CC88>
>         kw: List (size 0): {}
>         sleep_time: 1.96899999998
> Frame fire_timers in C:\Program Files\Python\27\lib\site-packages\eventlet\hubs\hub.py at line 462
>         heappop: <built-in function heappop>
>         self: <eventlet.hubs.selects.Hub object at 0x0000000003C8CC88>
>         when: 352173.859
>         timer: Timer(2.0, None, *None, **None)
>         next: List (size 2): (352173.859, Timer(2.0, None, *None, **None))
>         t: List (size 3): [(352173.89, Timer(2.0, <function _timeout at 0x0000000003FA4AC8>, *(timeout('timed o
>         exp: 352173.859
> Frame squelch_timer_exception in C:\Program Files\Python\27\lib\site-packages\eventlet\hubs\hub.py at line 387
>         exc_info: List (size 3): (<type 'exceptions.UnboundLocalError'>, UnboundLocalError("local variable 'from_addre
>         self: <eventlet.hubs.selects.Hub object at 0x0000000003C8CC88>
>         timer: Timer(2.0, None, *None, **None)

I'm not quite sure how I'd trigger this error again to test if the current github master fixes it, but I'll go ahead and install it.

For reference sake, I was using pypi's current version of eventlet/greenlet

> >pip show eventlet
> Name: eventlet
> Version: 0.21.0
> Summary: Highly concurrent networking library
> Home-page: http://eventlet.net
> Author: Linden Lab
> Author-email: eventletdev@lists.secondlife.com
> License: UNKNOWN
> Location: c:\program files\python\27\lib\site-packages
> Requires: greenlet, enum-compat
> 
> >pip show greenlet
> Name: greenlet
> Version: 0.4.12
> Summary: Lightweight in-process concurrent programming
> Home-page: https://github.com/python-greenlet/greenlet
> Author: Alexey Borzenkov
> Author-email: snaury@gmail.com
> License: MIT License
> Location: c:\program files\python\27\lib\site-packages
> Requires:

I will now install from the current master.

**Comment by MPagel** [comment]
I reinstalled them from their github repos, without apparent change in version.

>pip install -U https://github.com/eventlet/eventlet/archive/master.zip
Collecting https://github.com/eventlet/eventlet/archive/master.zip
  Downloading https://github.com/eventlet/eventlet/archive/master.zip
     / 819kB 581kB/s
Requirement already up-to-date: enum-compat in c:\program files\python\27\lib\site-packages (from eventlet==0.21.0)
Requirement already up-to-date: greenlet>=0.3 in c:\program files\python\27\lib\site-packages (from eventlet==0.21.0)
Requirement already up-to-date: enum34 in c:\program files\python\27\lib\site-packages (from enum-compat->eventlet==0.21.0)
Installing collected packages: eventlet
  Found existing installation: eventlet 0.21.0
    Uninstalling eventlet-0.21.0:
      Successfully uninstalled eventlet-0.21.0
  Running setup.py install for eventlet ... done
Successfully installed eventlet-0.21.0

>pip install -U https://github.com/python-greenlet/greenlet/archive/master.zip
Collecting https://github.com/python-greenlet/greenlet/archive/master.zip
  Downloading https://github.com/python-greenlet/greenlet/archive/master.zip
     - 92kB 708kB/s
Installing collected packages: greenlet
  Found existing installation: greenlet 0.4.12
    Uninstalling greenlet-0.4.12:
      Successfully uninstalled greenlet-0.4.12
  Running setup.py install for greenlet ... done
Successfully installed greenlet-0.4.12

Will consequently post this as a new issue.

**Comment by temoto** [comment]
Your traceback say

```
File "C:\Program Files\Python\27\lib\site-packages\eventlet\support\greendns.py", line 350, in end
raise result[1]
```

but in master that line 350 reads `return True`

is it possible you run old version?

Version number did not change indeed, I only raise version after PyPI release.

**Comment by MPagel** [comment]
due to a computer crash, I had to reinstall python and associated libraries, whereupon this error occurred again.  I was using the pypi greenlet/eventlet (0.4.12 and 0.21.0) after my reinstall.  Have now updated to 0.4.13.dev0 and 0.22.0 from your github repos.

(just pointing that it's an issue with the pypi hosted one circa a month or two ago; your current code likely addresses this problem completely).
